wallet/wallet_test.go

```diff
@@ -3512,3 +3513,115 @@ func TestEventTypes(t *testing.T) {
 		assertEvent(t, types.Hash256(types.SiafundOutputID(sfe[0].ID).V2ClaimOutputID()), wallet.EventTypeSiafundClaim, claimValue, types.ZeroCurrency, cm.Tip().Height+144)
 	})
 }
+
+func TestReset(t *testing.T) {
+	log := zaptest.NewLogger(t)
+	dir := t.TempDir()
+	db, err := sqlite.OpenDatabase(filepath.Join(dir, "walletd.sqlite3"), log.Named("sqlite3"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	bdb, err := coreutils.OpenBoltChainDB(filepath.Join(dir, "consensus.db"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer bdb.Close()
+
+	pk := types.GeneratePrivateKey()
+	addr := types.StandardUnlockHash(pk.PublicKey())
+
+	network, genesisBlock := testutil.Network()
+	// send the siafunds to the owned address
+	genesisBlock.Transactions[0].SiafundOutputs[0].Address = addr
+
+	store, genesisState, err := chain.NewDBStore(bdb, network, genesisBlock)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cm := chain.NewManager(store, genesisState)
+
+	wm, err := wallet.NewManager(cm, db, wallet.WithLogger(log.Named("wallet")), wallet.WithIndexMode(wallet.IndexModeFull))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer wm.Close()
+
+	// helper to mine blocks
+	mineBlock := func(n int, addr types.Address) {
+		t.Helper()
+		for i := 0; i < n; i++ {
+			b, ok := coreutils.MineBlock(cm, addr, 15*time.Second)
+			if !ok {
+				t.Fatal("failed to mine block")
+			} else if err := cm.AddBlocks([]types.Block{b}); err != nil {
+				t.Fatal(err)
+			}
+		}
+		waitForBlock(t, cm, db)
+	}
+
+	assertBalance := func(t *testing.T, addr types.Address, siacoin types.Currency, siafund uint64) {
+		t.Helper()
+
+		balance, err := db.AddressBalance(addr)
+		if err != nil {
+			t.Fatal(err)
+		} else if !balance.Siacoins.Equals(siacoin) {
+			t.Fatalf("expected %v SC, got %v", siacoin, balance.Siacoins)
+		} else if balance.Siafunds != siafund {
+			t.Fatalf("expected %v siafunds, got %v", siafund, balance.Siafunds)
+		}
+	}
+
+	// mine a payout to the original address
+	mineBlock(1, addr)
+
+	assertBalance(t, addr, types.ZeroCurrency, genesisState.SiafundCount())
+
+	// mine a bunch of payouts to random addresses
+	for i := 0; i < 50; i++ {
+		mineBlock(1, frand.Entropy256())
+	}
+
+	assertBalance(t, addr, genesisState.BlockReward(), genesisState.SiafundCount())
+
+	// close the wallet and reset it
+	if err := wm.Close(); err != nil {
+		t.Fatal(err)
+	} else if err := db.ResetChainState(); err != nil {
```

Comment:
Depending on how annoying it is to do my preference would be to make sure `NewManager` calls `ResetChainState` itself to also get that part into the test rather than manually resetting it.